README.md

```diff
@@ -1,19 +1,77 @@
 # Submit a GAR Public Site UI
 
-## Docker Usage
-The following depends on a .env file in the root of this project with the required environment variables. This will start a local test instance of keycloak for local testing.
+## Requirements
+- docker
+- docker-compose
+- node
+- npm
+- your email added to GOV Notify Dev whitelist by repository administrator
+
+## Development
+
+The following depends on a `.env.dev` file in the root of this project with
+the required environment variables. To obtain it please contact repository administrator.
+
+This will start a local instances of:
+- Database (used for both Frontend and Backend)
+- API (you need a local copy of `api` docker image loaded, 
+       please contact repository administrators to obtain it)
+- ClamAV engine
+- ClamAV REST API 
+- Frontend
+
+Please note - due to security configuration, the following lines need to be 
+commented out for local development, but not committed back to the repository:
+```
+./src/server.js (lines 95-98):
+  // app.use(csrf({cookie: {
+  //   httpOnly: true,
+  //   secure: true
+  // }}));
+
+./src/server.js (lines 103-106):
+    // var token = req.csrfToken();
+    // res.cookie('XSRF-TOKEN', token);
+    //
+    // res.locals.csrfToken = token;
+```
+
+
+To start clean:
+```
+docker-compose up --build -d
+```
+These steps will create 5 endpoints:
+- http://localhost:5432 -> Database
+- http://localhost:5000 -> API
+- http://localhost:3310 -> ClamAV engine
+- http://localhost:8080 -> ClamAV REST API
+- http://localhost:3000 -> Node site
+
+
+Once it's up, you need to:
+- whitelist yourself in database:
+```
+docker exec -ti database sh -c "PGPASSWORD=${POSTGRES_PASSWORD} psql -U ${POSTGRES_USER} ${POSTGRES_DB}"
```

Comment:
Coming from a Windows background I wasn't exactly sure what was going on (no egrep with Windows!) but got gist of what it was doing. A quick scan of [https://docs.docker.com/engine/examples/postgresql_service/](url) gave me a bit more info. But then again not being that UNIX savvy wasn't sure what psql was. I found that by installing PostgreSQL I could run the psql login command manually.

Reply:
@nigelhughes552 
So you need to export all environment variables to your local environment.
This will pass values to the `docker exec` command.
`docker exec` -> execute command in side docker 
`-ti database` -> do it in running image called `database` in interactive mode
`sh -c "PGPASSWORD=${POSTGRES_PASSWORD} psql -U ${POSTGRES_USER} ${POSTGRES_DB}"` -> command to execute (effectively connect to database using environment variables

`psql` command is executed inside running docker command, hence there is no need to have it installed locally (though as the Postgres port is exposed to localhost, you are able to connect to database via localhost:5432 address:port using any possible drivers)